lib/src/range/range.dart

```diff
@@ -246,9 +258,15 @@ class Range<T extends Comparable<T>> {
       if (!other.upperBounded) {
         return false;
       } else {
-        final upperCompare = Comparable.compare(upper, other.upper);
-        if (upperCompare < 0 || (upperCompare == 0 && !upperClosed &&
-            other.upperClosed)) {
+        if (upper == null) {
+          return false;
+        }
+        if (other.upper == null) {
```

Comment:
@hovadur here is the same: could you please combine the two checks above into one:

```dart
if (upper == null || other. upper == null) {
  return false;
}
```

Reply:
done